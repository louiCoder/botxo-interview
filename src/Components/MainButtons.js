import React from 'react';

export default function MainButtons ({ switchView, checked }) {
  return (
    <div className="main-buttons-container">
      <button onClick={() => switchView('ge')} className={checked === 'ge' ? 'active-main-button' : ''}>
        Get and Edit
      </button>
      <button onClick={() => switchView('ss')} className={checked === 'ss' ? 'active-main-button' : ''}>
        Show and send
      </button>
    </div>
  );
}
