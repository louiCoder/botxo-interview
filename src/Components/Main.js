import React, { Component, useState, useEffect } from 'react';
import GetAndEdit from './GetAndEdit';
import ShowAndSend from './ShowAndSend';
import './styles.scss';
import MainButtons from './MainButtons';
import SecondaryButtons from './SecondaryButtons';
import * as ACTION_CREATORS from '../Store/ActionCreators';
import { connect, useSelector, useDispatch } from 'react-redux';

export default () => {
  const [ mode, setMode ] = useState('ge');
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(ACTION_CREATORS.getAllUsersAction());
  }, []);

  const switchViewHandler = (mode) => setMode(mode);

  return (
    <div className="main-container">
      <div className="inner-container">
        <div className="top-container">
          <div className="first">
            <MainButtons switchView={(mode) => switchViewHandler(mode)} checked={mode} />
          </div>
          <div className="last">{mode === 'ge' ? <GetAndEdit /> : <ShowAndSend />}</div>
        </div>
        <div className="bottom-container">
          <SecondaryButtons switchView={(mode) => switchViewHandler(mode)} checked={mode} />
        </div>
      </div>
    </div>
  );
};

//
