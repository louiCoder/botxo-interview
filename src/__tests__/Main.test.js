import React from 'react';
import { mount, shallow } from 'enzyme';
import Main from '../Components/Main';
// import * as redux from 'react-redux';
// import Fixtures from '../Fixture';
import mockStore from '../Store/MockStore';
import { Provider } from 'react-redux';
import MainButtons from '../Components/MainButtons';

describe('<Main /> component', () => {
  let state;
  let wrapper;

  beforeEach(() => {
    const store = mockStore({ genericUsers: [], specificUsers: [], loading: false });
    const props = {
      switchView: jest.fn(),
      switchViewHandler: jest.fn()
    };
    wrapper = mount(
      <Provider store={store}>
        <Main {...props} />
      </Provider>
    );
  });

  it('Render without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with the right store state', () => {
    state = { genericUsers: [], specificUsers: [], loading: false };
    expect(wrapper.props().store.getState()).toEqual(state);
  });

  it('switches from one view to another', () => {
    const switchView = wrapper.find(MainButtons).prop('switchView');
    const button = wrapper.find(MainButtons).props();
    // console.log('debug', button);
    // console.log('wrapper', mainButtons);
    // const switchViewHandler = jest.fn();
    // const shal = mount(<Main />);
    // const spy = jest.spyOn(wrapper, 'switchViewHandler');
    // console.log('instance', spy);
  });
});
