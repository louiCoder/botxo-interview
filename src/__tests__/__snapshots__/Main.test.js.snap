// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> component Render without crashing 1`] = `ReactWrapper {}`;
